nautobot/utilities/testing/views.py

```diff
@@ -1404,3 +1404,45 @@ def test_bulk_add_component(self):
                 except AssertionError:
                     pass
             self.assertEqual(matching_count, self.bulk_create_count)
+
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+        def test_bulk_rename(self):
+            obj_perm = ObjectPermission(name="Test permission", actions=["change"])
+            obj_perm.save()
+            obj_perm.users.add(self.user)
+            obj_perm.object_types.add(ContentType.objects.get_for_model(self.model))
+
+            objects = self.selected_objects
+            pk_list = [obj.pk for obj in objects]
+            # Apply button not yet clicked
+            data = {"pk": pk_list}
+            data.update(self.rename_data)
+            verbose_name_plural = self.model._meta.verbose_name_plural
+
+            with self.subTest("Assert device name in HTML"):
+                response = self.client.post(self._get_url("bulk_rename"), data)
+                message = (
+                    f"Renaming {len(objects)} {self.selected_objects_parent_name} {bettertitle(verbose_name_plural)}"
+                )
+                self.assertInHTML(message, response.content.decode(response.charset))
+
+            with self.subTest("Assert update successfully"):
+                data["_apply"] = True  # Form Apply button
+                response = self.client.post(self._get_url("bulk_rename"), data)
+                self.assertHttpStatus(response, 302)
+                queryset = self._get_queryset().filter(pk__in=pk_list)
+                for instance in objects:
+                    self.assertEqual(queryset.get(pk=instance.pk).name, f"{instance.name}X")
+
+            with self.subTest("Assert if no objects selected return with error"):
+                data["pk"] = []
+                response = self.client.post(self._get_url("bulk_rename"), data, follow=True)
+                expected_message = f"""
+                <div class="alert alert-warning alert-dismissable" role="alert">
+                    <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+                        <span>&times;</span>
+                    </button>
+                    No {verbose_name_plural} were selected.
+                </div>
```

Comment:
Not sure it's in our best interest to check for the entire HTML block here, as it seems like that'll make the test pretty fragile as we update the UI. Maybe just check for `No {verbose_name_plural} were selected` instead?

Reply:
For that to work illd have to use `assertIn` ratter than `assertInHTML`, cause for some reason `assertInHTML` do not work